src/plugins/saved_objects_management/public/services/service_registry.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SavedObjectLoader } from '../../../saved_objects/public';
+
+export interface SavedObjectsManagementServiceRegistryEntry {
+  id: string;
+  service: SavedObjectLoader;
+  title: string;
+}
+
+export type ISavedObjectsManagementServiceRegistry = PublicMethodsOf<
+  SavedObjectsManagementServiceRegistry
+>;
+
+export class SavedObjectsManagementServiceRegistry {
+  private readonly registry = new Map<string, SavedObjectsManagementServiceRegistryEntry>();
+
+  public register(entry: SavedObjectsManagementServiceRegistryEntry) {
+    if (this.registry.has(entry.id)) {
+      throw new Error('');
+    }
+    this.registry.set(entry.id, entry);
+  }
+
```

Comment:
I did not adapt this one to the `Service` pattern, as I did for the `actions` service. Main reason was that this is supposedly going to be removed mid-term. Tell me if I still should do it.